go/cmd/dolt/commands/cvcmds/verify_constraints.go

```diff
@@ -109,51 +111,103 @@ func (cmd VerifyConstraintsCmd) Exec(ctx context.Context, commandStr string, arg
 	if err != nil {
 		return commands.HandleVErrAndExitCode(errhand.BuildDError("Unable to process constraint violations.").AddCause(err).Build(), nil)
 	}
-	if !outputOnly {
-		err = dEnv.UpdateWorkingRoot(ctx, endRoot)
-		if err != nil {
-			return commands.HandleVErrAndExitCode(errhand.BuildDError("Unable to update working root.").AddCause(err).Build(), nil)
-		}
+
+	err = dEnv.UpdateWorkingRoot(ctx, endRoot)
+	if err != nil {
+		return commands.HandleVErrAndExitCode(errhand.BuildDError("Unable to update working root.").AddCause(err).Build(), nil)
 	}
 
 	if tablesWithViolations.Size() > 0 {
 		cli.PrintErrln("All constraints are not satisfied.")
+		eng, err := engine.NewSqlEngineForEnv(ctx, dEnv)
+		if err != nil {
+			return commands.HandleVErrAndExitCode(errhand.BuildDError("Failed to build sql engine.").AddCause(err).Build(), nil)
+		}
+
 		for _, tableName := range tablesWithViolations.AsSortedSlice() {
-			table, ok, err := endRoot.GetTable(ctx, tableName)
+			tbl, ok, err := endRoot.GetTable(ctx, tableName)
 			if err != nil {
 				return commands.HandleVErrAndExitCode(errhand.BuildDError("Error loading table.").AddCause(err).Build(), nil)
 			}
 			if !ok {
 				return commands.HandleVErrAndExitCode(errhand.BuildDError("Unable to load table '%s'.", tableName).Build(), nil)
 			}
-			cvSch, err := table.GetConstraintViolationsSchema(ctx)
-			if err != nil {
-				return commands.HandleVErrAndExitCode(errhand.BuildDError("Error loading constraint violations schema.").AddCause(err).Build(), nil)
-			}
-			cvMap, err := table.GetConstraintViolations(ctx)
-			if err != nil {
-				return commands.HandleVErrAndExitCode(errhand.BuildDError("Error loading constraint violations data.").AddCause(err).Build(), nil)
-			}
-			sqlSchema, err := sqlutil.FromDoltSchema(tableName, cvSch)
-			if err != nil {
-				return commands.HandleVErrAndExitCode(errhand.BuildDError("Error attempting to convert schema").AddCause(err).Build(), nil)
-			}
-			rowIter, err := sqlutil.MapToSqlIter(ctx, cvSch, cvMap)
-			if err != nil {
-				return commands.HandleVErrAndExitCode(errhand.BuildDError("Error attempting to create row iterator").AddCause(err).Build(), nil)
-			}
 			cli.Println("")
 			cli.Println(doltdb.DoltConstViolTablePrefix + tableName)
-			if cvMap.Len() > 50 {
-				cli.Printf("Over 50 constraint violations were found. Please query '%s' to see them all.\n", doltdb.DoltConstViolTablePrefix+tableName)
-			} else {
-				err = engine.PrettyPrintResults(sql.NewEmptyContext(), engine.FormatTabular, sqlSchema.Schema, rowIter, false)
-				if err != nil {
-					return commands.HandleVErrAndExitCode(errhand.BuildDError("Error outputting rows").AddCause(err).Build(), nil)
-				}
+			dErr := printViolationsForTable(ctx, tableName, tbl, eng)
+			if dErr != nil {
+				return commands.HandleVErrAndExitCode(dErr, nil)
+			}
+		}
+
+		if outputOnly {
+			err = dEnv.UpdateWorkingRoot(ctx, working)
```

Comment:
Is there a good way for the engine to be initialized using a given working set? We want to temporarily add the violations if `--no-output` is set.

Reply:
DoltSession.UpdateWorkingSet?